server/filters.go

```diff
@@ -132,6 +132,75 @@ func pxcFilter(versions map[string]*pbVersion.Version, apply string, current str
 	return nil
 }
 
+func pgFilter(versions map[string]*pbVersion.Version, apply string, current string) error {
+	if len(versions) == 0 {
+		return status.Error(codes.Internal, "no versions to filter")
+	}
+
+	keys := make([]string, 0, len(versions))
+	for k, v := range versions {
+		if strings.ToLower(apply) == recommended && v.Status != pbVersion.Status_recommended {
+			continue
+		}
+
+		keys = append(keys, k)
+	}
+
+	sorted, err := sortedVersionsDesc(keys)
+	if err != nil {
+		return status.Errorf(codes.Internal, "failed to sort versions: %v", err)
+	}
+
+	if (strings.ToLower(apply) == recommended || strings.ToLower(apply) == latest) && current == "" {
+		return deleteOtherBut(sorted[0].String(), versions)
+	}
+
+	desired := apply //assume version number
```

Comment:
can we move this to beginning with
`desured := strings.ToLower(apply)`
coz at this place we call `ToLower` func 3 times already, and then 2 times more on the same value

Reply:
https://github.com/Percona-Lab/percona-version-service/pull/37/commits/61d255af3021eeea2d2912702e9edcaf779059df